x-pack/plugins/security_solution/server/endpoint/services/actions/create/types.ts

```diff
@@ -17,7 +16,7 @@ import type { ResponseActionsApiCommandNames } from '../../../../../common/endpo
 
 export type CreateActionPayload = TypeOf<typeof ResponseActionBodySchema> & {
   command: ResponseActionsApiCommandNames;
-  user?: ReturnType<AuthenticationServiceStart['getCurrentUser']>;
+  user?: { username: string } | null | undefined;
```

Comment:
Q: Should we just change `user` to be not optional for automated actions.

```suggestion
  user: { username: string } | undefined;
```

else just optional user should be okay?
```suggestion
  user?: { username: string } ;
```

Reply:
I'm not sure what the impact would be. for now, i'm thinking that we just leave it as is since I was not trying to change much about the existing logic. are you ok with that?

Reply:
Yeah that sounds okay about not changing too much. We can clean it up as we go later.